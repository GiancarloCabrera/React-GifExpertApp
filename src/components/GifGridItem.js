import React from 'react'
import PropTypes from 'prop-types';

export const GifGridItem = ({ img }) => {
  console.log('IMG:' ,img);
  return (
    <div className='card animate__animated animate__bounce'>
        <img src= { img.url } alt= { img.title }/>
        <p> { img.title }</p>
    </div>
  )
}

GifGridItem.propTypes = {
  img: PropTypes.object.isRequired
}
//ENZYME REQUIERE SIEMRPE QUE LOS COMPONENTES A EVALUAR SEAN EXPORTADOS 
// POR DEFAULT
export default GifGridItem;