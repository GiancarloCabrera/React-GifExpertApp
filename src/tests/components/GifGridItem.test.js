import React from "react";
import { shallow } from 'enzyme';
import GifGridItem from '../../components/GifGridItem';
describe('Pruebas en componente GifGridItem', () => {
    const obj = {
        title: 'un titulo',
        url: 'https://localhost:3000'
    };
    const app = shallow(
        <GifGridItem img={ obj }/>
    );

    test('Mostrar el componente de forma correcta', () => {
        expect( app ).toMatchSnapshot();
    });

    test('Debe tener un parrafo con el title', () => { 
        //p va a ser el parrado que se encuentre en GifGridItem
        const p = app.find('p');
        //Se espera que p (el parrafo), sin espacios al inicio y final,
        // sea igual a el titulo del objeto que se creo arriba
        expect(p.text().trim()).toBe(obj.title);       
    });

    test('Debe tener la imagen igual al url y alt de los props', () => { 
        const img = app.find('img');
        //Ver todas las propiedades
        console.log(img.props());
        //Ver una propiedad en especifico
        console.log(img.prop('src'));
        //Se espera que la img en su propiedad arc sea igual a la propiedad del objeto url
        expect(img.prop('src')).toBe(obj.url); 
        //Lo mismo con alt   
        expect(img.prop('alt')).toBe(obj.title);   
    });

    test('Debe tener la misma clase en el div', () => { 
        const div = app.find('div');
        const className = 'card animate__animated animate__bounce';
        expect(className).toBe(div.prop('className'));
     })
}) 