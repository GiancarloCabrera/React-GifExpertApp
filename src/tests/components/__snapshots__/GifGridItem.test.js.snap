// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente GifGridItem Mostrar el componente de forma correcta 1`] = `
<div
  className="card animate__animated animate__bounce"
>
  <img
    alt="un titulo"
    src="https://localhost:3000"
  />
  <p>
     
    un titulo
  </p>
</div>
`;
